packages/amplify-graphql-api-construct-tests/src/__tests__/deploy-velocity/deploy-velocity-test-core.ts

```diff
@@ -0,0 +1,216 @@
+      if (fs.existsSync(apiPostProcessorFilePath)) {
+        fs.unlinkSync(apiPostProcessorFilePath);
+      }
+      if (updatedApiPostProcessor) {
+        fs.writeFileSync(apiPostProcessorFilePath, generateApiPostProcessorFile(updatedApiPostProcessor));
+      }
+      const deployStartTimestamp = Date.now();
+      await cdkDeploy(projRoot, '--all', { timeoutMs: testDurationLimitMs });
+      const deployDurationMs = Date.now() - deployStartTimestamp;
+
+      // Verify Data Correctness Post-Deploy
+      if (dataValidate) {
+        await dataValidate(endpointConfig, setupState);
+      }
+      // Verify Deploy Speed
+      console.log(`Iterative Deploy Duration was ${deployDurationMs}ms for ${name}`);
+      expect(deployDurationMs).toBeLessThanOrEqual(testDurationLimitMs);
+
+      // Cleanup after test complete as part of test, if this hasn't run we'll run during `afterEach`.
+      // This has the side-effect of validating teardown in the test, but still attempting to cleanup if the test failed
+      await cdkDestroy(projRoot, '--all');
+      wasDestroyed = true;
+    });
+  });
+};
+
+/**
+ * Generate n fake uuids for testing
+ * @param count the number of ids to return
+ * @returns the list of fake ids
+ */
+export const generateFakeUUIDs = (count: number): string[] => Array.apply(0, new Array(count)).map((_) => faker.string.uuid());
+
+/**
+ * Given a list of object, and a chunk size, break into a list of lists, all of which has,
+ * at most, the number of elements provided by chunkSize.
+ * @param inputArray the input elements to partition
+ * @param chunkSize the max size of any given partition for the resulting arrays.
+ * @returns the input array partitioned by chunksize.
+ */
+export const splitArray = <T>(inputArray: T[], chunkSize: number): T[][] => {
+  if (chunkSize < 1) {
+    throw new Error('Chunk size must be greater than 0');
+  }
+  return inputArray.reduce((resultArray, item, index) => {
+    const chunkIndex = Math.floor(index / chunkSize);
+
+    if (!resultArray[chunkIndex]) {
+      resultArray[chunkIndex] = []; // start a new chunk
+    }
+
+    resultArray[chunkIndex].push(item);
+
+    return resultArray;
+  }, [] as T[][]);
+};
+
+const generateApiPostProcessorFile = (postProcessor: ApiPostProcessor): string => `module.exports = ${postProcessor.toString()}`;
+
+const getRecordCount = async (endpointConfig: EndpointConfig): Promise<number> => {
+  let nextToken = null;
+  let resultCount = 0;
+  do {
+    const response = await validateGraphql({
+      ...endpointConfig,
+      query: /* GraphQL */ `
+        query LIST_TODOS {
+          listTodos(nextToken: ${nextToken ? `"${nextToken}"` : 'null'}) {
+            items { id }
+            nextToken
+          }
+        }
+      `,
+      expectedStatusCode: 200,
+    });
+    nextToken = response.body.data.listTodos.nextToken;
+    resultCount += response.body.data.listTodos.items.length;
+  } while (nextToken);
+  return resultCount;
+};
+
```

Comment:
Could you add the comments for these util functions as they will be the core functions used in future test suites?